cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java

```diff
@@ -116,6 +118,7 @@ public abstract class AbstractIdeContext implements IdeContext {
   public AbstractIdeContext(IdeLogLevel minLogLevel, Function<IdeLogLevel, IdeSubLogger> factory, Path userDir) {
 
     super();
+    this.loggerFactory = factory;
     this.loggers = new HashMap<>();
     for (IdeLogLevel level : IdeLogLevel.values()) {
       IdeSubLogger logger;
```

Comment:
as you copy & pasted this code into `setLogLevel` method please replace it here with
```
setLogLevel(minLogLevel);
```

FYI: Github suggestion does not work out of reach from the actual code change...

Reply:
Adjusted now.